app.js

```diff
@@ -1,27 +1,135 @@
-// app.js
+const express = require('express');
+const mongoose = require('mongoose');
+const morgan = require('morgan');
+const cors = require('cors');
+const path = require('path');
+const multer = require('multer');
+const jimp = require('jimp');
+const fs = require('fs').promises;
+const { v4: uuidv4 } = require('uuid');
+require('dotenv').config();
 
-const express = require("express");
-const mongoose = require("mongoose");
-const morgan = require("morgan");
-const cors = require("cors");
-require("dotenv").config();
-
-const contactRoutes = require("./routes/api/contacts");
-const usersRoutes = require("./routes/api/users");
-const authenticate = require("./middlewares/authenticate");
+const contactRoutes = require('./routes/api/contacts');
+const userRoutes = require('./routes/api/users');
+const User = require('./models/user');
+const { isImageAndTransform } = require('./services/helper');
 
 const app = express();
 
+// Middleware
 app.use(express.json());
-app.use(morgan("dev"));
+app.use(morgan('dev'));
 app.use(cors());
+app.use('/avatars', express.static(path.join(__dirname, 'public/avatars')));
+
+// Multer configuration
+const storage = multer.diskStorage({
+  destination: function (req, file, cb) {
+    const tmpPath = path.join(__dirname, 'tmp'); // Temporary storage folder
+    fs.mkdir(tmpPath, { recursive: true })
+      .then(() => cb(null, tmpPath))
+      .catch(err => {
+        console.error('Error creating temporary directory:', err);
+        cb(err);
+      });
+  },
+  filename: function (req, file, cb) {
+    const uniqueSuffix = Date.now() + '-' + Math.round(Math.random() * 1E9);
+    cb(null, file.fieldname + '-' + uniqueSuffix + path.extname(file.originalname));
+  }
+});
+
+const upload = multer({ storage: storage });
+
+// Middleware for user authentication
+const authenticate = require('./middlewares/authenticate');
+
+app.patch('/api/users/avatars', authenticate, upload.single('avatar'), async (req, res) => {
+  try {
+    if (!req.file) {
+      console.log('No file uploaded');
+      return res.status(400).json({
+        status: 'error',
+        code: 400,
+        message: 'File is required',
+      });
+    }
+
+    if (!req.user || !req.user.id) {
```

Comment:
Nie rozumiem tego warunku tutaj :) Jeśli authenticate przepuści kod dalej to znaczy że user istnieje i ma dostęp do tego miejsca. Jeśli nie przepuści, czyli uwierzytelnienie zakończy się odrzuceniem usera to to miejsce się nie wieykona, podobnie upload.single.